### PR TITLE
adds uc processing

### PR DESCRIPTION
**Pull request by gregcaporaso**
@wasade, @audy I'm working on porting some code from QIIME 1 related to taxonomy assignment for use in a revision of [this paper](https://peerj.com/preprints/934/), and ultimately a QIIME 2 plugin. Are you ok with this repo being for all things taxonomy-related? We can add the sklearn classifiers in here too ultimately.

### CODE REVIEW

**Comment by wasade** [comment]
:+1: glad to help and review too!

**Comment by gregcaporaso** [comment]
Cool, thx. The initial version is going to be pretty sparse, just adding some API calls for uc to taxonomy processing (so we can adapt the uclust consensus assigner to be a vsearch consensus assigner). I'm going to push that in this morning.

**Comment by wasade** [comment]
Excellent, looking forward to seeing it

**Comment by gregcaporaso** [comment]
@wasade, this is ready for review if you're interested/available. There is only one public API function for now, and README documents how to use it. We're doing some experiments with this code - we can clean it up and add niceties if the package ends up going anywhere, but am not going to invest a ton of time in it at the moment in case our experiments don't pan out.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+sudo: false
+language: python
+env:
+  - PYTHON_VERSION=2.7
```
Comment (wasade) [inline .travis.yml]: Just to verify, we want py2? If so, then we won't be able to depend on skbio > 0.4.1, right?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,48 @@
...
+Consensus taxonomy assignments from .uc files
+---------------------------------------------
+
+This example illustrates how to apply the consensus taxonomy processing described in [Bokulich et al (in preparation)](https://peerj.com/preprints/934/). It assumes that you have the test data files included in this repository in ``./test-data``.
+
+```python
+import taxster
+
+taxonomy_map = {}
+for line in open('./test-data/uc/tax-map.tsv', 'U'):
```
Comment (wasade) [inline README.md]: totally minor, but would it be more concise to use pandas?

**Comment by wasade** [comment]
Only minor comments given the intent/focus

File: README.md (hunk 2 above)
Comment (audy) [inline README.md]: Pandas will read the entire file which can require a lot of memory. I only use it when I need to do actual data-frame stuff. You could use the csv parser in the standard library.

--austin

**Comment by audy** [comment]
I'm okay with this but it might be good to start with a clean master branch.

--austin

File: .travis.yml (hunk 1 above)
Comment (gregcaporaso) [inline .travis.yml]: I just need Py2 support temporarily for the taxonomy evaluation framework which I need to update to Py3. We can drop this once we need an skbio import (there's no skbio dependency at the moment).

**Comment by gregcaporaso** [comment]
@audy, there's not much in master (3 commits) so I'm not too concerned about it. If you feel strongly, could you let me know what you'd like me to do specifically?

File: README.md (hunk 2 above)
Comment (gregcaporaso) [inline README.md]: It's just a code example in the README so do you guys mind if I just leave as-is for now? I need to work on something else, and I'd like to just get it in.

Comment (wasade) [inline README.md]: no strong feelings

**Comment by wasade** [comment]
Just a note, I'm :+1: on merge

**Comment by gregcaporaso** [comment]
Thanks guys!

**Pull request merged**